crates/sui-core/src/consensus_adapter.rs

```diff
@@ -843,6 +892,72 @@ impl ConsensusAdapter {
             .inc();
     }
 
+    async fn submit_inner(
```

Comment:
This code is pretty sensitive. At a high level, why do we need to refactor them and what changed?